File: 1-adhock/4-check-ab/solution.js
function checkAB(str) {
  for (let i = 0; i < str.length; i++) {
    if ((str[i] == 'a' && str[i - 4] == 'b') || (str[i] == 'a' && str[i + 4] == 'b')) {
      return true
    }
  }
  return false
}

// Driver Code
console.log(checkAB('lane borrowed')) // True
console.log(checkAB('i am sick')) // False
console.log(checkAB('you are boring')) // True
console.log(checkAB('barbarian')) // True
console.log(checkAB('bacon and meat')) // False

File: 1-adhock/4-check-ab/solution.go
package main

import "fmt"

func checkAB(str string) bool {
	for i, _ := range str {
		if string(str[i]) == "a" {
			if i >= 4 {
				if string(str[i-4]) == "b" {
					return true
				}
			}
			if i <= len(str)-3 {
				if string(str[i+4]) == "b" {
					return true
				}
			}
		}
	}
	return false
}

func main() {
	fmt.Println(checkAB("lane borrowed"))  // True
	fmt.Println(checkAB("i am sick"))      // False
	fmt.Println(checkAB("you are boring")) // True
	fmt.Println(checkAB("barbarian"))      // True
	fmt.Println(checkAB("bacon and meat")) // False
}
